deployer/__main__.py

```diff
@@ -199,33 +233,76 @@ def validate(cluster_name):
+        required=True, dest="action", help="Available subcommands"
     )
-    subparsers = argparser.add_subparsers(dest="action")
 
-    deploy_parser = subparsers.add_parser("deploy")
-    validate_parser = subparsers.add_parser("validate")
-    deploy_support_parser = subparsers.add_parser("deploy-support")
-    deploy_grafana_parser = subparsers.add_parser("deploy-grafana")
+    # === Arguments and options shared across subcommands go here ===#
+    # NOTE: If you we do not add a base_parser here with the add_help=False
+    #       option, then we see a "conflicting option strings" error when
+    #       running `python deployer --help`
+    base_parser = argparse.ArgumentParser(add_help=False)
+    base_parser.add_argument(
+        "cluster_name",
+        type=str,
+        help="The name of the cluster to authenticate against and perform actions on",
+    )
 
-    deploy_parser.add_argument("cluster_name")
-    deploy_parser.add_argument("hub_name", nargs="?")
-    deploy_parser.add_argument("--skip-hub-health-test", action="store_true")
+    # === Add new subcommands in this section ===#
+    # Deploy subcommand
+    deploy_parser = subparsers.add_parser(
+        "deploy",
+        parents=[base_parser],
+        help="Install/upgrade the helm charts of JupyterHubs on a cluster",
+    )
+    deploy_parser.add_argument(
+        "hub_name",
+        nargs="?",
+        help="The hub, or list of hubs, to install/upgrade the helm chart for",
+    )
+    deploy_parser.add_argument(
+        "--skip-hub-health-test", action="store_true", help="Bypass the hub health test"
+    )
     deploy_parser.add_argument(
         "--config-path",
-        help="Read deployment config from this file",
+        help="File to read secret deployment configuration from",
         default="config/secrets.yaml",
     )
 
-    validate_parser.add_argument("cluster_name")
+    # Validate subcommand
+    validate_parser = subparsers.add_parser(
+        "validate",
+        parents=[base_parser],
+        help="Validate the cluster configuration against a JSON schema",
+    )
+
+    # deploy-support subcommand
+    deploy_support_parser = subparsers.add_parser(
+        "deploy-support",
+        parents=[base_parser],
+        help="Install/upgrade the support helm release on a given cluster",
+    )
 
-    deploy_support_parser.add_argument("cluster_name")
+    # deploy-grafana-dashboards subcommand
+    deploy_grafana_dashboards_parser = subparsers.add_parser(
+        "deploy-grafana-dashboards",
+        parents=[base_parser],
+        help="Deploy grafana dashboards to a cluster for monitoring JupyterHubs. deploy-support must be run first!",
+    )
 
-    deploy_grafana_parser.add_argument("cluster_name")
+    # Auth subcommand
+    auth_parser = subparsers.add_parser(
+        "auth",
+        parents=[base_parser],
+        help="Authenticate against a cluster using deployer credentials to gain quick command line access",
```

Comment:
Authenticate against a cluster sounds like its actually setting something up against the remote cluster. I think this command setups credentials for k8s cluster access, but I'm not sure what it does. Is it only modifying the kubeconfig (i think so)? Is it influencing environment variables (i don't think so)? I hope to see have this be more explicit on what is done.